text/3493-precise-pre-release-cargo-update.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: precise-pre-release-cargo-update
+- Start Date: 2023-09-20
+- RFC PR: [rust-lang/rfcs#3493](https://github.com/rust-lang/rfcs/pull/3493)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes extending `cargo update` to allow updates to pre-release versions when requested with `--precise`.
+For example, a `cargo` user would be able to call `cargo update -p dep --precise 0.1.1-pre0` as long as the version of `dep` requested by their project and its dependencies are semver compatible with `0.1.1`.
```

Comment:
I would suggest changing the examples throughout this to use `-pre.0` like versions, to avoid the issues with `-pre2 > -pre10`.

Reply:
:+1: https://github.com/rust-lang/rfcs/pull/3493/commits/fbc31fad7cfd7cedd88b9d77f48c1570e8b4b343